### ISSUE TITLE
LibreNMS submits zeros to Graphite for missing metrics

### ISSUE THREAD

**Issue by ktims** [librenms/librenms#14258]
### The problem

LibreNMS sends 0 to Graphite for missing datapoints, rather than omitting them. This causes issues as the validity of the data is no longer retained. Zero is a valid measurement and shouldn't be sent unless LibreNMS has actually collected a zero from the monitored device, so missing data can continue to be handled correctly in downstream data processing.

I believe this is a result of #7166 / #7176.

```
// Send zero for fields without values
if (empty($v)) {
   $v = 0;
}
```

Should just `continue` here instead to skip the metric.

### Output of ./validate.php

```txt
====================================
Component | Version
--------- | -------
LibreNMS  | 1.69-1-gbc02ab3
DB Schema | 2020_07_27_00522_alter_devices_snmp_algo_columns (188)
PHP       | 7.2.31
Python    | 3.6.8
MySQL     | 5.5.65-MariaDB
RRDTool   | 1.4.8
SNMP      | NET-SNMP 5.7.2
OpenSSL   | 
====================================
```

**Comment by murrant** [comment]
Please submit a change request.  However, the if check should probably be `if (is_null($v)) {`

**Comment by ktims** [comment]
I've attempted to repro this with a fresh test instance, and it looks like Libre is doing the right thing when the data source is unreachable. I reviewed the relevant code between 1.69.1 in prod and the test instance and I can't find much difference in the data storage logic. So I am not quite sure what happened here, but at first glance it looks like the flawed logic here is not at issue.

In any case, the logic definitely does seem flawed to me, so I will submit a PR.
